src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,181 @@
+        boolean proceedOnwards = false;
+        try {
+            taskNum = Integer.parseInt(command.substring(5));
+            proceedOnwards = true;
+        } catch (StringIndexOutOfBoundsException e) {
+            printErrorMessage("Please provide a valid number.");
+        }
+
+        if (proceedOnwards) {
+            try {
+                tasks.get(taskNum - 1).setDone();
+                System.out.println(Duke.horizLine);
+                System.out.println("Nice! I've marked this task as done:\n" + "[\u2713] " + tasks.get(taskNum - 1).getDescription());
+                System.out.println(Duke.horizLine);
+            } catch (ArrayIndexOutOfBoundsException e) {
+                printErrorMessage("Number given is not in the list.");
+            } catch (IndexOutOfBoundsException e) {
+                printErrorMessage("Please provide a valid number.");
+            }
+        }
+    }
+
+    private static void printList(ArrayList<Task> tasks) {
+        System.out.println(Duke.horizLine);
+        System.out.println("Here are the tasks on your list: ");
+        for (int i = 0; i < tasks.size() ; i++) {
+            System.out.println((i+1) + ". " + tasks.get(i).toString());
+        }
+        System.out.println(Duke.horizLine);
+    }
+
+    private static void computeEventCommand(ArrayList<Task> tasks, String command) {
+        String[] eventDetails = new String[2];
+        try {
+            extractCommandDetails(eventDetails, command, "Event");
+            printAcknowledgement();
+            tasks.add(new Event(eventDetails[0], eventDetails[1]));
+            System.out.println(tasks.get(tasks.size() - 1).toString());
+            System.out.println("There are now " + tasks.size() + " tasks in the list.");
+            System.out.println(Duke.horizLine);
+        } catch (StringIndexOutOfBoundsException e) {
+            printErrorMessage("Description for an Event cannot be left empty.");
+        }
+    }
+
+    private static void computeDeadlineCommand(ArrayList<Task> tasks, String command) {
+        String[] deadlineDetails = new String[2];
+        try {
+            extractCommandDetails(deadlineDetails, command, "Deadline");
+            printAcknowledgement();
+            tasks.add(new Deadline(deadlineDetails[0], deadlineDetails[1]));
+            System.out.println(tasks.get(tasks.size() - 1).toString());
+            System.out.println("There are now " + tasks.size() +  " tasks in the list.");
+            System.out.println(Duke.horizLine);
+        } catch (StringIndexOutOfBoundsException e) {
+            printErrorMessage("Description for a Deadline cannot be left empty.");
+        }
+    }
+
+    private static void extractCommandDetails(String[] commandDetails, String command, String commandType)
+            throws StringIndexOutOfBoundsException{
+        int dividerPos = command.indexOf('/');
+        commandDetails[0] = command.substring((commandType.equals("Deadline") ? 9 : 6), dividerPos);
```

Comment:
Maybe you should avoid using magic literals like `9` or `6`, instead you can declare a constant